README.md

```diff
@@ -161,6 +161,34 @@ will be added as they are required.
 
 ---
 
+`should_not_redirect`
```

Comment:
I've looked this over and I'm still not clear on when you would use `should_not_redirect`.  The way that all of the other functions have been written is to assert the state that you _want_ to happen, not the states that you _don't want_ to happen.  I'm thinking instead of using `should_not_redirect` you would instead use something like `should_select_pool` or `should_discard`.

Can you please elaborate for me?  Maybe give me an example of two?